node/node.go

```diff
@@ -170,73 +172,64 @@ type Node struct {
 	prometheusSrv    *http.Server
 }
 
-// NewNode returns a new, ready to go, Tendermint Node.
-func NewNode(config *cfg.Config,
-	privValidator types.PrivValidator,
-	nodeKey *p2p.NodeKey,
-	clientCreator proxy.ClientCreator,
-	genesisDocProvider GenesisDocProvider,
-	dbProvider DBProvider,
-	metricsProvider MetricsProvider,
-	logger log.Logger) (*Node, error) {
-
-	// Get BlockStore
+func getStoresForNewNode(config *cfg.Config, dbProvider DBProvider) (*bc.BlockStore, dbm.DB, error) {
 	blockStoreDB, err := dbProvider(&DBContext{"blockstore", config})
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	blockStore := bc.NewBlockStore(blockStoreDB)
-
-	// Get State
 	stateDB, err := dbProvider(&DBContext{"state", config})
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
+	return blockStore, stateDB, nil
+}
 
+func loadStateFromDBOrGenesisDoc(stateDB dbm.DB, genesisDocProvider GenesisDocProvider) (sm.State, *types.GenesisDoc, error) {
 	// Get genesis doc
 	// TODO: move to state package?
```

Comment:
can we move this to `state/store.go`?

Reply:
We can. It does, however, involve moving the `GenesisDocProvider`-related code from `node/node.go` to `state/store.go` to prevent circular dependencies. This changes the Go API a bit. Do you think that'd be a problem?

Technically the genesis doc represents a form of state, so it would make sense to have it in the `state` package, right?

Reply:
Hmm.. will respond tomorrow